sdjournal/read.go

```diff
@@ -188,6 +190,83 @@ process:
 	return
 }
 
+func (r *JournalReader) FollowWithFields(until <-chan time.Time, fields map[string]struct{}, writer io.Writer) (err error) {
+	enc := json.NewEncoder(writer)
+journal:
+	for {
+		kvMap := make(map[string]string)
+		select {
+		case <-until:
+			return ErrExpired
+		default:
+			var err error
+			var c int
+
+			// Advance the journal cursor
+			c, err = r.journal.Next()
+
+			// An unexpected error
+			if err != nil {
+				return err
+			}
+
+			// We have a new journal entry go over the fields
+			// get the data for what we care about and return
+			r.journal.RestartData()
+			if c > 0 {
+			fields:
+				for {
+					s, err := r.journal.EnumerateData()
+					if err != nil || len(s) == 0 {
+						break fields
+					}
+					s = s[:len(s)]
```

Comment:
Actually in retrospect, i think this method should be called `FollowWithAllFields` because many people end up using custom fields in their journal entries. This allows the user to have the option to filter or not.

I think the best way of doing so would be taking out the `fields map[string]struct{}` parameter to the function